sympy/calculus/util.py

```diff
@@ -162,7 +162,7 @@ def function_range(f, symbol, domain):
         return S.EmptySet
 
     period = periodicity(f, symbol)
-    if period is S.Zero:
+    if period == S.Zero:
```

Comment:
It looks like `S.Zero` is the only result from `periodicity` that could satisfy this.